docs/mca/set-mca-param.rst

```diff
@@ -0,0 +1,251 @@
+
+             shell$ export OMPI_MCA_param="value with multiple words"
+
+.. warning:: Setting Open MPI MCA parameters via environment variables
+             entails prefixing the parameter name with ``OMPI_MCA_``.
+             When setting PMIx- and PRRTE-specific MCA parameters via
+             environment variables, use a different prefix:
+
+             +----------+----------------+
+             | Open MPI | ``OMPI_MCA_``  |
+             +----------+----------------+
+             | PMIx     | ``PMIX_MCA_``  |
+             +----------+----------------+
+             | PRRTE    | ``PRRTE_MCA_`` |
+             +----------+----------------+
+
+.. _label-mca-set-param-tune:
+
+Tuning MCA parameter files
+--------------------------
+
+.. error:: TODO This entire section needs to be checked for correctness.                                                
+
+Simple text files can be used to set MCA parameter values for a
+specific application.
+
+The ``mpirun --tune`` CLI option allows users to specify both MCA
+parameters and environment variables from within a single file.
+
+MCA parameters set in tuned parameter files will override any MCA
+parameters supplied in global parameter files (e.g.,
+``$HOME/.openmpi/mca-params.conf``), but not command line or
+environment parameters.
+
+Consider a tuned parameter file name ``foo.conf`` that is placed in
+the same directory as the application ``a.out``. A user will typically
+run the application as:
+
+.. code-block:: sh
+
+   shell$ mpirun -np 2 a.out
+
+To use the ``foo.conf`` tuned parameter file, this command line
+changes to:
+
+.. code-block:: sh
+
+   shell$ mpirun -np 2 --tune foo.conf a.out
+
+Tuned parameter files can be coupled if more than one file is to be
+used. If there is another tuned parameter file called ``bar.conf``, it
+can be added to the command line as follows:
+
+.. code-block:: sh
+
+   shell$ mpirun -np 2 --tune foo.conf,bar.conf a.out
+
+The contents of tuned files consist of one or more lines, each of
+which contain zero or more `-x` and `--mca` options.  Comments are not
+allowed.  For example, the following tuned file:
+
+.. code-block::
+
+   -x envvar1=value1 -mca param1 value1 -x envvar2
+   -mca param2 value2
+   -x envvar3
+
+is equivalent to:
+
+.. code-block:: sh
+
+   shell$ mpirun \
+       -x envvar1=value1 -mca param1 value1 -x envvar2 \
+       -mca param2 value2
+       -x envvar3 \
+       ...rest of mpirun command line...
+
+Although the typical use case for tuned parameter files is to be
+specified on the command line, they can also be set as MCA parameters
+in the environment.  The MCA parameter ``mca_base_envvar_file_prefix``
+contains a comma-delimited list of tuned parameter files exactly as
+they would be passed to the ``--tune`` command line option.  The MCA
+parameter ``mca_base_envvar_file_path`` specifies the path to search
+for tuned files with relative paths.
+
+.. error:: TODO Check that these MCA var names ^^ are correct.
```

Comment:
Can you check this?

Reply:
Okay will do

Reply:
Looks like they're not correct and also mca_base_envar_files is a deprecated synonym, fixing